src/components/product.js

```diff
@@ -461,8 +472,7 @@ export default class Product extends Component {
    * @return {String}
    */
   get onlineStoreURL() {
-    const identifier = this.handle ? this.handle : this.id;
-    return `https://${this.props.client.config.domain}/products/${identifier}${this.onlineStoreQueryString}`;
+    return `${this.model.onlineStoreUrl}${this.onlineStoreQueryString}`;
```

Comment:
it adds query params to the url such as the referring channel / domain, as well as which specific variant is chosen